validator/project_validator.go

```diff
@@ -157,6 +157,7 @@ var projectWarningValidators = []projectValidator{
 	checkTaskRuns,
 	checkModules,
 	checkTasks,
+	checkTaskDependencies,
```

Comment:
The functionality of this function was extracted from `validateTaskDependencies`. I didn't change the implementation for either, I did remove the corresponding dead code from both since it is a split.